app/assets/javascripts/services/CountryService.js

```diff
@@ -0,0 +1,202 @@
+define([
+  'Class',
+  'uri',
+  'bluebird',
+  'map/services/DataService'
+], function(Class, UriTemplate, Promise, ds) {
+
+  'use strict';
+
+  var CONFIG = {
+    countriesConfigDataset: '134caa0a-21f7-451d-a7fe-30db31a424aa',
+    countriesConfigTable: 'gfw_countries_config',
+    countriesDataset: '134caa0a-21f7-451d-a7fe-30db31a424aa',
+    countriesTable: 'gadm28_countries',
+    regionsDataset: '098b33df-6871-4e53-a5ff-b56a7d989f9a',
+    regionsTable: 'gadm28_adm1',
+    subRegionsDataset: 'b3d076cc-b150-4ccb-a93e-eca05d9ac2bf',
+    subRegionsTable: 'gadm28_adm2'
+  };
+
+  var GET_REQUEST_COUNTRY_CONFIG_ID = 'CountryService:getCountries',
+      GET_REQUEST_COUNTRIES_LIST_ID = 'CountryService:getCountries',
+      SHOW_REQUEST_COUNTRY_ID = 'CountryService:showCountry',
+      GET_REQUEST_REGIONS_LIST_ID = 'CountryService:getRegionsList',
+      SHOW_REQUEST_REGION_ID = 'CountryService:showRegion';
+
+  var APIURL = window.gfw.config.GFW_API_HOST_PROD;
+
+  var APIURLS = {
+    'getCountryConfig'   : '/query/{countriesConfigDataset}?sql=SELECT iso, indepth FROM {countriesConfigTable} WHERE iso=\'{iso}\'',
+    'getCountriesList'   : '/query/{countriesDataset}?sql=SELECT name_engli as name, iso FROM {countriesTable}',
+    'showCountry'        : '/query/{countriesDataset}?sql=SELECT name_engli as name, iso, topojson FROM {countriesTable} WHERE iso=\'{iso}\'',
+    'getRegionsList'     : '/query/{regionsDataset}?sql=SELECT id_1, name_1 FROM {regionsTable} WHERE iso=\'{iso}\' ORDER BY name_1',
+    'showRegion'         : '/query/{regionsDataset}?sql=SELECT id_1, name_1, geojson FROM {regionsTable} WHERE iso=\'{iso}\' AND id_1={region} ORDER BY name_1',
+    'getSubRegionsList'  : '/query/{subRegionsDataset}?sql=SELECT id_1, name_1 FROM {subRegionsTable} WHERE iso=\'{iso}\' ORDER BY name_1'
+  };
+
+  var CountriesService = Class.extend({
```

Comment:
Niiiice service! Much better to have all of the countries/regions logic together 😁